command/agent/consul/service_client.go

```diff
@@ -1760,6 +1760,15 @@ func (c *ServiceClient) AllocRegistrations(allocID string) (*serviceregistration
 					sreg.Checks = append(sreg.Checks, check)
 				}
 			}
+
+			if sidecarService := services[serviceID+sidecarSuffix]; sidecarService != nil {
```

Comment:
this feels like a bit of a mouthful, so I went to double-check it, and I found this thing:

https://github.com/hashicorp/nomad/blob/d9c9dc548a6037d9e71acdb8b717fd8784a1f581/command/agent/consul/service_client.go#L2123-L2130

nit-picky, but I think I prefer having the magic-string key encapsulated in that one spot:

```suggestion
			if sidecarService := getNomadSidecar(serviceID, services); sidecarService != nil {
```

Reply:
Good catch :+1: